activerecord/lib/active_record/associations/collection_association.rb

```diff
@@ -301,6 +301,15 @@ def find_from_target?
           target.any? { |record| record.new_record? || record.changed? }
       end
 
+      def insert_all_attributes
+        join_keys = reflection.join_keys
+        key = join_keys.key
+        foreign_key = join_keys.foreign_key
+        value = owner[foreign_key]
+
+        { key => value }
```

Comment:
I'm out of my depth here, but this doesn't seem right to me. I've asked @kamipo to check this.

If we this is right, we should cut out some local variables:

```ruby
{ reflection.join_keys.key => owner[reflection.join_keys.foreign_key] }
```

Reply:
Thanks, @kaspth . Fixed this.